web/controllers/v1/memory_service_controller.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Cog.V1.MemoryServiceController do
+  use Cog.Web, :controller
+  require Logger
+
+  plug Cog.Plug.ServiceAuthentication
+  import Cog.Plug.Util, only: [get_service_token: 1]
+
+  alias Cog.Command.Service.Memory
+
+  def show(conn, %{"key" => key}) do
+    service_token = get_service_token(conn)
```

Comment:
Feels like this check should be done in a plug for two reasons:

1. DRY up duplicate token checking logic
2. Return a more accurate status code (400/401/422) instead of 500 in the case of an invalid token.